sdk/go/pulumi/context.go

```diff
@@ -814,6 +838,26 @@ func (ctx *Context) InvokePackageRaw(
 	return hasSecret, nil
 }
 
+func (ctx *Context) InvokePackageRawWithDeps(
```

Comment:
In https://github.com/pulumi/pulumi/pull/17132/files#diff-6ac1847413cb55c8d480423da6e23ed6ac98c0066fcf7c57f8b5cd960a04dee1R662 we added `InvokePackageRaw` as an alternative to `InvokePackage` that also returns an `isSecret` bool.

We now add a 3rd variant `InvokePackageRawWithDeps` that additionally returns the dependencies so that we can add them to the output wrapped result on the SDK side.

Reply:
It's too bad we have to keep adding new named methods. Should we consider adding a result struct and "one last" new method that returns the result struct, to give us flexibility to add other results in the future, if needed, without having to add a new method?

If so, what would we name the method and type?

Reply:
Python returns an `InvokeResult`. What's a little confusing with go though is that the actual result value  is an argument to the function (`result interface{}`, the function return value only holds "metadata".

Reply:
This was still bothering me and I kept tinkering with while traveling on the train this weekend, and I finally realised that instead of continuing on the path started with https://github.com/pulumi/pulumi/pull/17132, we can push the Output creation down into the Go SDK, and introduce a `InvokeOutput` method.